### ISSUE TITLE
ASViewController - LayoutSpecThatFits Protocol

### ISSUE THREAD

**Issue by Reedyuk** [TextureGroup/Texture#1076]
Feature request for the Swift purists out there.
It grinds me that currently when i have an ASViewController with a simple displayNode as the root node, then i have to manually do this:

```
node.layoutSpecBlock = { (node, constrainedSize) in
      return ASInsetLayoutSpec(insets: UIEdgeInsets.zero, child: self.tableNode)
}
```

I think that for the purist in me, would prefer to override the method in my implementation of the ASViewController, like so:

```
override func layoutSpecThatFits(_ constrainedSize: ASSizeRange) -> ASLayoutSpec {
      return ASInsetLayoutSpec(insets: UIEdgeInsets.zero, child: self.tableNode)
}
```

I'm not sure if anybody else has this thought or maybe its just me. Anyway, i would be interested if this could be implemented and if not, how come.

P.S. ASInsetLayoutSpec was a crap example for a single layout spec.

**Comment by hashemp206** [comment]
`LayoutSpecThatFits ` is part `ASDisplayNode` behavior. I alway create display node and set it to `ASViewController`

layout of display nodes is not responsibility of ViewController

**Comment by Reedyuk** [comment]
I disagree. In the view controller you still will end up doing a layoutSpecBlock if you need to layout that view, unless you have a custom displayNode object that performs the layout in itself.
The item i'm debating is the fact that the layoutSpecBlock is performed in the view controller init method, so why not have a convenience method in the view controller to do this instead of a block.
